feat: Create first routes, completion of the missing routes

MattConce opened the pull request:
Signed-off-by: Matheus Santos <mattsantos.conce@gmail.com>

closes #24

MattConce committed:
feat: Create first routes, completion of the missing routes

Signed-off-by: Matheus Santos <mattsantos.conce@gmail.com>

ricardokojo commented on package.json:
```diff
@@ -22,7 +22,8 @@
     "express": "^4.17.1",
-    "mongodb": "^3.6.2"
+    "mongodb": "^3.6.2",
+    "body-parser": "*"
```
É bom deixar uma versao fixa da dependência, para não ficar atualizando toda hora

ricardokojo commented on src/test.sh:
```diff
@@ -0,0 +1,14 @@
+printf "POST: create new user \n"
```
Para criar os testes, é melhor usar o Jest mesmo

MattConce replied:
Isso foi só para mostrar o que eu tinha testado mesmo. Tira do commit esse arquivo?

ricardokojo commented on src/routes/routes.js:
```diff
@@ -21,5 +21,50 @@ router.get("/:cpf", mongoMiddleware, async (req, res) => {
 });
 
+router.post("/", mongoMiddleware, async (req, res) => {
+  const user = {name: req.body.name, cpf: req.body.cpf, email: req.body.email,
```
Quebrar linhas

ricardokojo commented on src/routes/routes.js:
```diff
@@ -21,5 +21,50 @@ router.get("/:cpf", mongoMiddleware, async (req, res) => {
+    const response = await req.dbConnection
+      .collection("records")
+      .updateOne({"cpf":cpf}, 
+        {$set: {"email": email}}, {upsert: true});
```
Atualizar outros dados também

ricardokojo commented on src/routes/routes.js:
```diff
@@ -21,5 +21,50 @@ router.get("/:cpf", mongoMiddleware, async (req, res) => {
+    const response = await req.dbConnection
+      .collection("records")
+      .deleteOne({"cpf": cpf});
+    console.log(cpf);
```
Remover console.log

MattConce commented:
Fiz um rebase pra tirar o commit antigo, @AlexiaScheffer precisa revisar outra vez poque acabou dando dismissed

vini-gpereira committed:
refactor: update data structure

Signed-off-by: Ricardo Kojo <ricardokojo7@gmail.com>
Signed-off-by: Thais Lasso <thlasso@usp.br>
Signed-off-by: Vinicius Pereira <vini.gpereira2014@gmail.com>

vini-gpereira committed:
chore: add validator to post route

Signed-off-by: Ricardo Kojo <ricardokojo7@gmail.com>
Signed-off-by: Thais Lasso <thlasso@usp.br>
Signed-off-by: Vinicius Pereira <vini.gpereira2014@gmail.com>

vini-gpereira committed:
chore: update patch route using new data structure
chore: verify if patientId exists on database

Signed-off-by: Ricardo Kojo <ricardokojo7@gmail.com>
Signed-off-by: Thais Lasso <thlasso@usp.br>
Signed-off-by: Vinicius Pereira <vini.gpereira2014@gmail.com>

vini-gpereira committed:
chore: update get using new data structure

Signed-off-by: Ricardo Kojo <ricardokojo7@gmail.com>
Signed-off-by: Thais Lasso <thlasso@usp.br>
Signed-off-by: Vinicius Pereira <vini.gpereira2014@gmail.com>

vini-gpereira committed:
chore: update delete route to new data structure
chore: add missing status return on routes

Signed-off-by: Ricardo Kojo <ricardokojo7@gmail.com>
Signed-off-by: Thais Lasso <thlasso@usp.br>
Signed-off-by: Vinicius Pereira <vini.gpereira2014@gmail.com>

vini-gpereira committed:
chore: clear null values on patch route insertion

Signed-off-by: Ricardo Kojo <ricardokojo7@gmail.com>
Signed-off-by: Vinicius Pereira <vini.gpereira2014@gmail.com>

vini-gpereira committed:
feat: add initial data validation to routes
fix: patch updates data with null values
chore: stop routes execution on error with void return

Signed-off-by: Ricardo Kojo <ricardokojo7@gmail.com>
Signed-off-by: Vinicius Pereira <vini.gpereira2014@gmail.com>
Signed-off-by: Wander Souza <wanderdasouza@gmail.com>
